vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -0,0 +1,293 @@
+	vulnDir := filepath.Join(directory, cnaAssigner)
+	err := os.MkdirAll(vulnDir, 0755)
+	if err != nil {
+		Logger.Warnf("Failed to create dir: %v", err)
+		return fmt.Errorf("failed to create dir: %w", err)
+	}
+
+	var fileWriteErr error
+
+	// Only write out the OSV file if we have ranges.
+	outputFile := filepath.Join(vulnDir, v.ID+extension)
+
+	f, err := os.Create(outputFile)
+	if err != nil {
+		notes = append(notes, fmt.Sprintf("Failed to open %s for writing: %v", outputFile, err))
+		fileWriteErr = err
+	} else {
+		err = v.ToJSON(f)
+		if err != nil {
+			notes = append(notes, fmt.Sprintf("Failed to write %s: %v", outputFile, err))
+			fileWriteErr = err
+		} else {
+			Logger.Infof("[%s]: Generated OSV record under the %s CNA", CVE.Metadata.CVEID, cnaAssigner)
+		}
+		f.Close()
+	}
+
+	Metrics.Notes = append(Metrics.Notes, notes...)
+	metricsFile := filepath.Join(vulnDir, v.ID+".metrics.json")
+	if marshalledMetrics, err := json.MarshalIndent(Metrics, "", "  "); err != nil {
+		Logger.Warnf("[%s]: Failed to marshal metrics: %v", CVE.Metadata.CVEID, err)
+	} else {
+		if err = os.WriteFile(metricsFile, marshalledMetrics, 0660); err != nil {
+			Logger.Warnf("[%s]: Failed to write %s: %v", CVE.Metadata.CVEID, metricsFile, err)
+		}
+	}
+
+	if fileWriteErr != nil {
+		return fileWriteErr
+	}
+
+	return nil
+}
+
+// identifyPossibleURLs extracts all possible URLs from a CVE5 object,
+// including those in CNA and ADP containers, and affected package information.
+// It deduplicates the URLs before returning them.
+func identifyPossibleURLs(cve cves.CVE5) []cves.Reference {
+	refs := cve.Containers.CNA.References
+	for _, adp := range cve.Containers.ADP {
+		if adp.References != nil {
+			refs = append(refs, adp.References...)
+		}
+	}
+
+	for _, affected := range cve.Containers.CNA.Affected {
+		if affected.CollectionUrl != "" {
+			refs = append(refs, cves.Reference{Url: affected.CollectionUrl})
+		}
+		if affected.Repo != "" {
+			refs = append(refs, cves.Reference{Url: affected.Repo})
+		}
+	}
+
+	// Remove duplicate URLs
+	slices.SortStableFunc(refs, func(a, b cves.Reference) int {
+		return strings.Compare(a.Url, b.Url)
+	})
+	refs = slices.CompactFunc(refs, func(a, b cves.Reference) bool {
+		return a.Url == b.Url
+	})
+	return refs
+}
+
+func main() {
+	flag.Parse()
+	if !slices.Contains([]string{"OSV"}, *outFormat) {
+		fmt.Fprintf(os.Stderr, "Unsupported output format: %s\n", *outFormat)
+		os.Exit(1)
+	}
+
+	var logCleanup func()
+	Logger, logCleanup = utility.CreateLoggerWrapper("cvelist-osv")
+	defer logCleanup()
+	data, err := os.ReadFile(*jsonPath)
+	if err != nil {
+		Logger.Fatalf("Failed to open file: %v", err) // double check this is best practice output
+	}
+
+	var cve cves.CVE5
+	err = json.Unmarshal(data, &cve)
+	if err != nil {
+		Logger.Fatalf("Failed to parse CVEList CVE JSON: %v", err)
+	}
+
+	ReposForCVE := make(map[cves.CVEID][]string)
+	refs := identifyPossibleURLs(cve)
+
+	CVEID := cve.Metadata.CVEID
+
```

Comment:
Hmm... doesn't the notes already have the CVE ID prefixed? (At least looking at the changes you made below it looks like it would already have the CVE ID.

Either way, I think this should be done in ReposFromReferencesCVEList, not something the caller should do.